Data/Classes/RecipeData.lua

```diff
@@ -545,16 +545,16 @@ end
 ---@param options? CraftSim.RecipeData.OptimizeProfitOptions
 function CraftSim.RecipeData:OptimizeProfit(options)
     options = options or {}
-    if options.optimizeReagents then
-        self:OptimizeReagents()
-    end
-    if options.optimizeGear then -- TEMPORARY: ALWAYS OPTIMIZE IF PARAM IS NOT GIVEN
-        self:OptimizeGear(CraftSim.TOPGEAR:GetSimMode(CraftSim.TOPGEAR.SIM_MODES.PROFIT))
-    end
     if options.optimizeReagents and options.optimizeGear then
-        -- need another because it could be that the optimized gear makes it possible to use cheaper reagents
+        -- Optimize gear first
+        self:OptimizeGear(CraftSim.TOPGEAR:GetSimMode(CraftSim.TOPGEAR.SIM_MODES.PROFIT))

```

Comment:
Thats a tricky one

Before it was
reagents -> gear -> reagents

My thought was that reagent optimization always aims for higher result quality of the recipe (disregarding profit)

But Gear always aims for highest profit here

so for example the first reagent optimizations pushes us past a breakpoint
Then the gear optimization tells us to use multicraft gear cause now u dont need inspiration anymore
but then another reagent optimization could consider the newly gained skill from the optimized gear


actually i think this is a situation where we could just optimize reagents -> optimize gear in a loop 
and it just gets more precise the more you loop with a fixed break after a few attempts ^^
but yeah performance is always a problem here since the gear optimization takes a lot of computing time

whats your thought on this?
I think we should at least keep the reagent -> gear -> reagent thing

Reply:
Hey !
I think you're right.
I  didn't think about that  loop thing.
Maybe it's possible to optimize the process in another version ;)